website_sale_b2x_alt_price/tests/test_ui.py

```diff
@@ -5,120 +5,131 @@
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(
+            context=dict(
+                cls.env.context,
+                mail_create_nolog=True,
+                mail_create_nosubscribe=True,
+                mail_notrack=True,
+                no_reset_password=True,
+                tracking_disable=True,
+            )
+        )
         # Create and select a pricelist
         # to make tests pass no matter what l10n package is enabled
-        website = self.env["website"].get_current_website()
-        pricelist = self.env["product.pricelist"].create(
+        website = cls.env["website"].get_current_website()
+        pricelist = cls.env["product.pricelist"].create(
             {
                 "name": "website_sale_b2x_alt_price public",
                 "currency_id": website.user_id.company_id.currency_id.id,
                 "selectable": True,
             }
         )
         website.user_id.property_product_pricelist = pricelist
-        admin = self.env.ref("base.user_admin")
+        admin = cls.env.ref("base.user_admin")
         admin.property_product_pricelist = pricelist
         # Create some demo taxes
-        self.tax_group_22 = self.env["account.tax.group"].create(
+        cls.tax_group_22 = cls.env["account.tax.group"].create(
             {"name": "Tax group 22%"}
         )
-        self.tax_22_sale = self.env["account.tax"].create(
+        cls.tax_22_sale = cls.env["account.tax"].create(
             {
                 "amount_type": "percent",
                 "amount": 22,
                 "description": "22%",
                 "name": "Tax sale 22%",
-                "tax_group_id": self.tax_group_22.id,
+                "tax_group_id": cls.tax_group_22.id,
                 "type_tax_use": "sale",
             }
         )
-        self.tax_22_purchase = self.env["account.tax"].create(
+        cls.tax_22_purchase = cls.env["account.tax"].create(
             {
                 "amount_type": "percent",
                 "amount": 22,
                 "description": "22%",
                 "name": "Tax purchase 22%",
-                "tax_group_id": self.tax_group_22.id,
+                "tax_group_id": cls.tax_group_22.id,
                 "type_tax_use": "purchase",
             }
         )
         # Create one product without taxes
-        self.training_accounting = self.env["product.template"].create(
+        cls.training_accounting = cls.env["product.template"].create(
             {
                 "name": "Training on accounting - website_sale_b2x_alt_price",
                 "list_price": 100,
                 "type": "service",
                 "website_published": True,
-                "uom_id": self.ref("uom.product_uom_unit"),
-                "uom_po_id": self.ref("uom.product_uom_unit"),
+                "uom_id": cls.ref("uom.product_uom_unit"),
+                "uom_po_id": cls.ref("uom.product_uom_unit"),
```

Comment:
(some others to check)
```suggestion
                "uom_id": cls.env.ref("uom.product_uom_unit"),
                "uom_po_id": cls.env.ref("uom.product_uom_unit"),
```